lib/logeater/request.rb

```diff
@@ -6,14 +6,9 @@ class Request < ActiveRecord::Base
     self.table_name = "requests"
 
     def self.import(values)
-      # values have to be unique by uuid
-      values = values.uniq { |request| request.uuid }
-
-      super values
-    rescue PG::UniqueViolation, ActiveRecord::RecordNotUnique
-      # try again after skipping requests with existing uuids
+      # values have to be unique by uuid so we pull already saved ones out before trying to insert
       existing_uuids = where(uuid: values.map(&:uuid)).pluck(:uuid)
-      values = values.reject { |request| existing_uuids.member?(request.uuid) }
+      values = values.uniq { |request| request.uuid }.reject { |request| existing_uuids.member?(request.uuid) }
```

Comment:
This might be the only flag you want to take a look at. I couldn't figure out how to recover from `PG::UniqueViolation`. Hitting that marks the transaction as failed and you can't even do lookups like `where(uuid: values.map(&:uuid)).pluck(:uuid)` anymore. I didn't really see a difference from what it was to how it is now where we just preemptively remove the duplicates instead of recovering, then removing them. :man_shrugging:

Reply:
Alright! This change makes sense to me 👍

Reply:
⬇️ 👔 `.uniq { |request| request.uuid }` could be `uniq(&:uuid)`